design/src/rfcs/rfc0023_serialization_and_deserialization.md

```diff
@@ -0,0 +1,90 @@
+<!-- Give your RFC a descriptive name saying what it would accomplish or what feature it defines -->
+RFC: Serialization and Deserialization
+=============
+
+<!-- RFCs start with the "RFC" status and are then either "Implemented" or "Rejected".  -->
+> Status: RFC
+>
+> Applies to: builder structs and other data types that are connected to builder structs
+
+# Overview
+There has been some github issues requesting for the implementation of Serialize/Deserialize traits provided serde crate.  
+At the current state of sdk, user must implement their own solution in order to serialize/deserialize sdk's datatypes.  
+This is a RFC for implementing serde traits.
+
+# Potential Use Cases
+- Testing
+- Reusable conifiguration, infrastructure as a service
+- Logging
+ 
+... and more!
+
+# Implementation
+Add `serde::Serialize` and `serde::Deserialize` to builder datatypes.
+
+# Discussion
+## Enum Representation 
+serde allows programmers to use one of four different tagging ([internal, external, adjacent and untagged](https://serde.rs/enum-representations.html)) when serializing enum.  
+
+I suggest going with `untagged` variant.  
+This way serialized data can be used with aws-cli's file-based input.  
```

Comment:
Is this input specified anywhere? It would be good to reference it in the RFC so that the future implementer (probably you) can write tests ensuring it works.

Reply:
I messed it up.

I mistakenly believed that AWS CLI's file based input is automatically generated from smithy.

I couldn't find anything that backs me up.

Let me delete this section.

Reply:
I think compatibility with the CLI would still be cool, maybe we could do this with a later feature that could read the same format? In cases like this, it's nice to say that we want this to be possible but that we leave it to some later RFC to define how.